### DISCUSSION TITLE
Console Error - Extension ms-python.python CANNOT USE these API proposals...

### DISCUSSION THREAD

**Discussion by chiaolinghong** [CodinGame/monaco-vscode-api#414] [Q&A] [answered]
Hi, I've modified the **demo** files to get a cleaner version of the editor. However, I am getting this Error console log once the python extension has been installed.

Here's the updated **constructOptions** **(IWorkbenchConstructionOptions)** in `setup.commons.ts`:
- Changed the extensionsGallery serviceUrl since the one provided in the demo seems to be outdated. (I want the latest python extension with proper limited web-supports for Python and Pylance)
- Add a `additionalBuiltinExtensions` to install python on initial load, which seems to work as expected. (though sometimes requires a page refresh)
```
export const constructOptions: IWorkbenchConstructionOptions = {
  ...,
  productConfiguration: {
    extensionsGallery: {
      serviceUrl: 'https://open-vsx.org/vscode/gallery',
      itemUrl: 'https://open-vsx.org/vscode/item',
      resourceUrlTemplate: 'https://open-vsx.org/vscode/unpkg/{publisher}/{name}/{version}/{path}',
      controlUrl: '',
      nlsBaseUrl: '',
      publisherUrl: ''
    }
  },
  additionalBuiltinExtensions: ['ms-python.python']
}
```
![Extension CANNOT USE these API proposals](https://github.com/CodinGame/monaco-vscode-api/assets/101141561/bed6d11e-f5cc-4968-ad79-e64b0252d3a9)

Any insights or help on how to fix it or proper way to install Python extension automatically on initial load would be much appreciated. Thanks.

**Comment by CGNonofr** [comment]
> Changed the extensionsGallery serviceUrl since the one provided in the demo seems to be outdated.

Outdated? It just use an open source extension registry instead of the microsoft one

> Add a additionalBuiltinExtensions to install python on initial load, which seems to work as expected. (thoguh sometimes requires a page refresh)

Only builtin extensions can access api proposals, and installing it that way doesn't consider the extension as builtin

You better install and import `@codingame/monaco-vscode-python-default-extension`

**Reply by chiaolinghong** [reply]
@CGNonofr Thanks for answering,
So I have the default-python-extension installed but that is just basic syntax highlighting.
Only the microsoft one has web support for the Python and Pylance. If you open up the extension gallery in the demo site, it does not allow installing Python extension which includes Pylance (what I need)
I've also tried using `commands.executeCommand`
```
commands.executeCommand('workbench.extensions.installExtension', 'ms-python.python')
```
Which show up the same console error along with a pop up prompt to acknowledge the limitation on web version of Python.

The main goal I am trying to achieve is to install the python extension, or any extension from the gallery, without any user interactions. Is this possible with installing extensions from the Extension Gallery?

**Reply by CGNonofr** [reply, marked as answer]
You can put the vsix file in your project and import it via the vsix-plugin

**Reply by CGNonofr** [reply]
see https://github.com/CodinGame/monaco-vscode-api?tab=readme-ov-file#loading-vsix-file

**Reply by chiaolinghong** [reply]
Got it :) Thanks!
